org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/InstallationManagement.scala

```diff
@@ -158,13 +160,22 @@ trait InstallationManagement { this: ScalaProject =>
     projectSpecificStorage.addPropertyChangeListener(compilerSettingsListener)
   }
 
+  private lazy val actualAdditionalParams: AtomicReference[() => String] = new AtomicReference(() => {
+    actualAdditionalParams.getAndSet(() => storage.getString(CompilerSettings.ADDITIONAL_PARAMS))
```

Comment:
not sure maybe it is too protective but idea is to set it atomic. `set` API does not guarantee it.